### PR TITLE
Icons0

### LINKED ISSUE #14: main_icon
Add the main icon, that changes with the current weather.

### PR DESCRIPTION
**Pull request by catalinac3**
closes #14 
- resolve error in data.rain[1h], I have inserted an extra “rain“ and added units of mm.
- Added main icon using the icons offer by open weather API, main icon should change with current weather
- Imported awesome icons. Change svg icon from bootstrap to font awesome icon on the rain parameter.
- In the rain parameter: if is raining there is an umbrella, if not a beach umbrella icon.
- updated README.

### CODE REVIEW

File: jsFiles/app.js [hunk 1]
```diff
@@ -72,16 +74,24 @@ function getTemperature(latitude, longitude) {
...
       // when the weather is not rainy data.rain doesn't exits
-      rainVolume.innerHTML = (data.rain) ?  data.rain.rain["1h"]:"no rain"
+      rainVolume.innerHTML = data.rain ? `${data.rain["1h"]}mm` : "no rain";
+      rainIcon.className = data.rain
+        ? "fas fa-umbrella"
+        : "fas fa-umbrella-beach";
     })
     .catch((error) => {
       // handles errors
       console.warn(`error: ${error.message}, weather API`);
```
Comment (MrTim) [inline jsFiles/app.js]: as a next step you can create a proper error handling - display what happened to the user in a popup for example.

File: jsFiles/countries_code.js [hunk 2]
```diff
@@ -245,8 +245,9 @@ const isoCountries = {
   ZM: "Zambia",
   ZW: "Zimbabwe",
 };
-// This function converts the code of a country to the country name 
+// This function converts the code of a country to the country name
 // if it doesn't exist, it returns the code itself
 function countryCodeConversion(countryCode) {
+  // bracket notation is needed because countryCode is a variable
   return countryCode in isoCountries ? isoCountries[countryCode] : countryCode;
```
Comment (MrTim) [inline jsFiles/countries_code.js:252]: you can actually make it even shorter `return isoCountries[countryCode] || countryCode`.

And the even more shorter function will looks like:
```js
const countryCodeConversion = (countryCode) => (isoCountries[countryCode] || countryCode);
```
But you can stick to the classic version of creating a function :)

**Review by MrTim — APPROVED** [review]
Great stuff! left some comments in places which you can improve.

File: jsFiles/countries_code.js (hunk 2 above)
Comment (catalinac3) [reply jsFiles/countries_code.js:252]: Wao, that is very short, maybe for the moment, I'll stick to the classic.

File: jsFiles/app.js (hunk 1 above)
Comment (catalinac3) [reply jsFiles/app.js]: I could add in this pull request a function that makes a pop up with an alert (errror.message) like used in getLocation() and use it for both handleError() and catch()....
On a later pull request I can add something like the bootstrap alert: https://getbootstrap.com/docs/4.0/components/alerts/.

Comment (MrTim) [reply jsFiles/app.js]: yep, that's a good way to go!

File: jsFiles/app.js [hunk 3]
```diff
@@ -98,3 +99,8 @@ function timeConversion(timeStamp) {
     minute: "2-digit",
   });
 }
+
+// handles the alert pop up message for catch and handle error functions
+function alertError(error){
```
Comment (MrTim) [inline jsFiles/app.js]: space between `) {`

File: jsFiles/app.js [hunk 4]
```diff
@@ -98,3 +99,8 @@ function timeConversion(timeStamp) {
...
+  return alert(error.message)
```
Comment (MrTim) [inline jsFiles/app.js]: don't omit semicolon after parentheses

File: jsFiles/app.js [hunk 5]
```diff
@@ -35,8 +35,8 @@ function getLocation(pos) {
 }
 
 function handleError(error) {
-  alert(`Your position could not be found, error: ${error.message}`);
-  console.warn(`Error displaying position, error: ${error.code}`);
+  alertError(error);
+  console.warn(`${error.message}, error: ${error.code}`);
```
Comment (MrTim) [inline jsFiles/app.js]: if you have this alert, then you don't need console :)

**Review by MrTim — CHANGES_REQUESTED** [review]

**Review by MrTim — APPROVED** [review]
:+1: 
Please merge it to master!

**Pull request merged**